src/core/domain-services/crux-gateway.ts

```diff
@@ -8,29 +9,41 @@ import {
 } from "../interfaces";
 
 export class CertificateManager {
-    public static make = (idClaim: IGatewayIdentityClaim): IGatewayIdentityCertificate => {
+    public static make = async (idClaim: IGatewayIdentityClaim, messageId: string): Promise<IGatewayIdentityCertificate> => {
+        const payload = {
+            messageId,
+        };
+        const signedProof = await idClaim.keyManager.signWebToken(payload);
         return {
-            claim: idClaim.cruxId.toString(),
-            proof: "PROOF",
+                claim: idClaim.cruxId.toString(),
+                messageId,
+                proof: signedProof,
         };
     }
     public static verify = (certificate: IGatewayIdentityCertificate) => {
-        return true;
+        const proof: any = decodeToken(certificate.proof).payload;
```

Comment:
you decoded token, but you also have to verify the signature

Reply:
Callers of `verify` do not expect an Error if there is something wrong with certificate (while decoding/verifying). Just return false in those cases

Reply:
> you decoded token, but you also have to verify the signature

verification for signature added.

Reply:
> Callers of `verify` do not expect an Error if there is something wrong with certificate (while decoding/verifying). Just return false in those cases

try-catch block added for this kind of scenarios